socially/view/ajax_files/data.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+global $i;
+require_once 'controller/post_controller/Posts.php';
+require_once 'model/Query.php';
+require 'controller/post_controller/get_post_time.php';
+
+// Takes the starting value from where the next posts will be shown to the user.
+$start = $_POST['starting'];
+$posts = new Posts();
+$date_time = new DateTime('now', new DateTimeZone('Asia/Kolkata'));
+$curr_time = strtotime($date_time->format('Y-m-d H:i:s'));
+// Loads next 5 posts from start index.
+$data = $posts->fetch_posts($start, 5);
+$i += $start + 1;
+foreach ($data as $item => $datum) {
+  $user = $datum['user'];
+  $text = $datum['text'];
+  $post = $datum['post'];
+  $post_time = strtotime($datum['post_time']);
+  $pfp = '';
+  $output = '';
+  $time_diff = get_post_time($curr_time,$post_time);
+  // If a profile photo exists, display that, else show the default blank user
+  // image.
+  if (file_exists("view/profile_photos/{$user}.jpeg")) {
+    $pfp = "view/profile_photos/{$user}.jpeg";
+  }
+  else {
+    $pfp = 'assets/user.png';
+  }
+  $output .= '<div class="post" data-post-id="'. $i++ . '">
+  <div class="top">
+    <img src=' . $pfp . '><a href="/profile/' . $user .
+    '" class="username">' . $user . '</a>
+    <p class="post_time">' . $time_diff . '</p>
+  </div>
+  <p class="post-text">' . $text .
+    '</p>';
+  $post = 'view/' . $post;
+  $output .= '<div class="media">';
+  // Checks the post-media type before displaying to the web page.
+  if ($post !== '' && $post != null) {
+    if (str_starts_with(mime_content_type($post), 'image')) {
+      $output .= '<div class="post-img">
+                  <img src="' . $post . '">
+              </div>';
+    }
+    elseif (str_starts_with(mime_content_type($post), 'audio')) {
+      $output .= '<div class="post-audio">
+                  <audio controls src="' . $post . '"></audio>
+              </div>';
+    }
+    elseif (str_starts_with(mime_content_type($post), 'video')) {
+      $output .= '<div class="post-video">
+        <video controls loop width="660px" src="' . $post  . '"></video>
+      </div>';
+    }
+  }
+  $output .= '</div> <br>
+  <div class="interactions">
```

Comment:
Since we are generating html directly inside of php this can lead to maintenance challenges later. So can we refactor the code ?

Reply:
here do I have to return the response as json instead of returning html as string?

Reply:
Yes that we can do.
Additionally we can store this html content in a template file and use that too

Reply:
okay